src/org/zaproxy/zap/extension/ascanrulesAlpha/MongoDbInjection.java

```diff
@@ -0,0 +1,494 @@
+					String textBase = getBaseMsg().getResponseBody().toString();
+					String textInj = msgInj.getResponseBody().toString();
+					if(!textBase.equals(textInj)) {
+						// Get more confidence
+						if(getMoreConfidence) {
+							HttpMessage secondMsg = getNewMsg();
+							setParameter(secondMsg, param+URL_TOKEN, valueInj);
+							sendAndReceive(secondMsg, false);
+							String secondText = secondMsg.getResponseBody().toString();
+							if(!textInj.equals(secondText) && !areDifferentForOnlyValue(secondText, textInj, URL_TOKEN.
+									substring(1, URL_TOKEN.length()-1), paramInj.substring(1, paramInj.length()-1))) {
+								bingo(Alert.RISK_HIGH, Alert.CONFIDENCE_HIGH, getName(), getDescription(), null, param,
+										paramInj+valueInj, getExtraInfo(ALL_DATA_ATTACK) , getSolution(), msgInj);
+								isBingo = true;
+								break;
+							}
+						}
+						else {
+							bingo(Alert.RISK_HIGH, Alert.CONFIDENCE_MEDIUM, getName(), getDescription(), null, param,
+									paramInj+valueInj, getExtraInfo(ALL_DATA_ATTACK) , getSolution(), msgInj);
+							isBingo = true;
+							break;
+						}
+					}
+				}catch(IOException ex) {
+					if (LOG.isDebugEnabled()) {
+						LOG.debug("Caught " + ex.getClass().getName() + " " + ex.getMessage() +" when "+URI_EX_LOG);
+					}					
+					return;
+				}
+			}
+		}
+		// not-handled error injection
+		if(doCrashScan) {
+			if (LOG.isDebugEnabled()) {
+				LOG.debug("\nStarting with the not-handled error injection package of attack rules:");
+			}
+			for(String valueInj : CRASH_INJECTION) {
+				if(isStop()) {
+					if (LOG.isDebugEnabled()) { LOG.debug("Stopping the scan due to a user request"); }
+					return;
+				}
+				if (LOG.isDebugEnabled()) {
+					LOG.debug("\nTrying with the value: "+valueInj);
+				}
+				try {
+					HttpMessage msgInj = getNewMsg();
+					setParameter(msgInj, param, valueInj);
+					sendAndReceive(msgInj, false);
+					rtt.add(msgInj.getTimeElapsedMillis());
+					for(Pattern pattern : errorPatterns) {
+						Matcher matcher =  pattern.matcher(msgInj.getResponseBody().toString());
+						if(matcher.find()) {
+							bingo(Alert.RISK_HIGH, Alert.CONFIDENCE_MEDIUM, getName(), getDescription(), null, param,
+									valueInj, getExtraInfo(CRASH_ATTACK), getSolution(), msgInj);
+							isBingo = true;
+							break;
+						}
+					}
+				}catch(IOException ex) {
+					if (LOG.isDebugEnabled()) {
+						LOG.debug("Caught " + ex.getClass().getName() + " " + ex.getMessage() +" when "+URI_EX_LOG);
+					}
+					return;
+				}
+			}
+		}
+		// javascript code injection
+		if(doTimedScan) {
+			if(isStop()) {
+				if (LOG.isDebugEnabled()) { LOG.debug("Stopping the scan due to a user request"); }
+				return;
+			}
+			if (LOG.isDebugEnabled()) {
+				LOG.debug("\nStarting with the javascript code injection package of attack rules:");
+			}
+			int aveRtt = getAveRtts(rtt),
+				index = 0,
+				// the mongo sleep function multiplies by 3 the value (in ms) passed

```

Comment:
Is this documented somewhere? Could you add a link?

Reply:
I had the empirical certainty that in version 3.4.10 was done so, however, to ensure I tried to contact the MongoDB team and I'm investigating in person in their repository. 
As soon as I resolve I will give you confirmation.

Reply:
@LuigiCasciaro Any news on this?

Reply:
@kingthorin No, for the moment. In the official documentation (https://docs.mongodb.com/manual/reference/method/sleep/) this is not (obviously) reported. Anyway, this weekend I will do some new tests and try to resolve the question.

Reply:
@LuigiCasciaro Any news on this particular detail?